### ISSUE TITLE
Using vscode-go from multiple user accounts clashes on /tmp/go-code-check

### ISSUE THREAD

**Issue by lmb** [Microsoft/vscode-go#1829]
Steps to Reproduce:

1. On a system where TMPDIR is shared for the whole machine (e.g. Ubuntu)
2. Create two user accounts A and B
3. Work on Go code with A
4. Switch to user B, work on Go code
5. Observe error message

```
go build package/name: open /tmp/go-code-check: permission denied
```

```
$ ls -lh /tmp/go-code-check
-rw-rw-r-- 1 private private 173K Aug  7 22:54 /tmp/go-code-check
```

I believe this is due to https://github.com/Microsoft/vscode-go/blob/a1b8648a0b2ef5858669fa648a438f79c668f256/src/goBuild.ts#L59 assuming that TMP is exclusive to a user, which isn't the case on a default Ubuntu install.

**Comment by lmb** [comment]
Option 1: generate a random subdirectory, see 
https://golang.org/src/io/ioutil/tempfile.go?s=2138:2195#L66

Option 2: include the UID in the temp file name.

**Comment by ramya-rao-a** [comment]
Thanks for logging this issue @lmb and Thanks for the PR @rikkuness! Your fix is now available in the latest update (0.6.86) to the Go extension

**Comment by lmb** [comment]
Thank you both for the quick fix!
